### PR TITLE
Add in documentation of new alterUFFields Hook

### PR DESCRIPTION
**Pull request by seamuslee001**
ping @eileenmcnaughton

### CODE REVIEW

File: docs/hooks/hook_civicrm_alterUFFields.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+This hook allows for the modification of the avaliable fields that are permissiable for use within a profile.
+
+## Definition
+
+    hook_civicrm_alterUFFIelds(&$fields) {
+
+## Parameters
+
+-   $fields - a multidimential array of Entities and their fields.
+      For each field contained within the array there are at the following properties
```
Comment (eileenmcnaughton) [inline docs/hooks/hook_civicrm_alterUFFields.md]: @seamuslee001 which, if any, of these properties are used?

File: docs/hooks/hook_civicrm_alterUFFields.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+-   null
+
+## Example
+
+```php
+  /**
+   * Implementation of hook_civicrm_alterUFFIelds
+   */
+  function mte_civicrm_alterUFFIelds(&$fields) {
+    // Incldue grant fields in the permissable array
```
Comment (eileenmcnaughton) [inline docs/hooks/hook_civicrm_alterUFFields.md]: sp

File: docs/hooks/hook_civicrm_alterUFFields.md (hunk 1 above)
Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterUFFields.md]: As far as I know all are I believe

File: docs/hooks/hook_civicrm_alterUFFields.md (hunk 2 above)
Comment (MikeyMJCO) [reply docs/hooks/hook_civicrm_alterUFFields.md]: Is that the stray capital I you're pointing to?

**Comment by MikeyMJCO** [comment]
Looks good to me other than the capital I in the naming.

File: docs/hooks/hook_civicrm_alterUFFields.md (hunk 2 above)
Comment (eileenmcnaughton) [reply docs/hooks/hook_civicrm_alterUFFields.md]: I could Incldue that too....

Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterUFFields.md]: both should be fixed now

File: docs/hooks/hook_civicrm_alterUFFields.md (hunk 1 above)
Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterUFFields.md]: I know for certain that the name is and i think others are used but i think the general gist is that your generating the same output as from a `entity.getfields` api call

File: docs/hooks/hook_civicrm_alterUFFields.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# hook_civicrm_alterUFFIelds
+
+## Summary
+
+This hook allows for the modification of the available fields that are permissible for use within a profile.
```
Comment (eileenmcnaughton) [inline docs/hooks/hook_civicrm_alterUFFields.md]: Can we just flesh this out a bit more

Comment (seamuslee001) [reply docs/hooks/hook_civicrm_alterUFFields.md]: I have fleshed it out a bit more now @eileenmcnaughton

File: docs/hooks/hook_civicrm_alterUFFields.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+-   null
+
+## Example
+
+```php
+  /**
+   * Implementation of hook_civicrm_alterUFFIelds
+   */
+  function mte_civicrm_alterUFFields(&$fields) {
+    // Incldue grant fields in the permissible array
```
Comment (eileenmcnaughton) [inline docs/hooks/hook_civicrm_alterUFFields.md]: Still Incldue ....

**Comment by eileenmcnaughton** [comment]
Thanks @seamuslee001  - @MikeyMJCO  - I think is good

**Pull request merged**